### PR TITLE
Add abs, round, floor, ceil operators

### PR DESCRIPTION
**Pull request by anandthakker**
Port of https://github.com/mapbox/mapbox-gl-js/pull/6496

### CODE REVIEW

File: src/mbgl/style/expression/compound_expression.cpp [hunk 1]
```diff
@@ -362,6 +363,11 @@ std::unordered_map<std::string, CompoundExpressionRegistry::Definition> initiali
         return result;
     });
     
+    define("round", [](double x) -> Result<double> { return { std::round(x) }; });
```
Comment (jfirebaugh) [inline src/mbgl/style/expression/compound_expression.cpp]: The extra brackets in the return expression should be unnecessary.

**Comment by jfirebaugh** [comment]
Are we going to include SDK bindings for these in boba?

**Comment by 1ec5** [comment]
> https://github.com/mapbox/mapbox-gl-native/pull/11653#issuecomment-380298696

`abs`, `floor`, and `ceil` all correspond to built-in NSExpression functions: `abs:`, `floor:`, and `ceiling:`, respectively. The iOS/macOS SDK already synthesizes these functions based on existing operators. We can easily remove the circumlocution here:

https://github.com/mapbox/mapbox-gl-native/blob/33d3c27ab039e77f7d5691b1040b8b2af9615ec4/platform/darwin/src/NSExpression%2BMGLAdditions.mm#L829-L840

`round` has no native equivalent in NSExpression, but as of #11472, all JSON expression operators are automatically representable in NSExpression via the `MGL_FUNCTION()` function. It’d be pretty straightforward to add a dedicated `mgl_round:` function as syntactic sugar.

**Comment by 1ec5** [comment]
d30fbb48ab25d52b43ed73799f6377b5a56bd768 and 8aa1c1e32d9a95b57f5ccfe4e0662b1fccfc5b73 take care of all the changes needed in the iOS and macOS map SDKs.

**Comment by tobrun** [comment]
Android integration in https://github.com/mapbox/mapbox-gl-native/pull/11655 was approved and has been merged into this PR.

**Comment by anandthakker** [comment]
Thanks @tobrun @1ec5 !

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/layers/PropertyFactory.java [hunk 2]
```diff
@@ -2046,7 +2046,7 @@
   }
 
   /**
-   * Value to use for a text label. Within literal values and zoom functions, property names enclosed in curly brackets (e.g. `{token}`) are replaced with the value of the named property. Expressions and property functions do not support this syntax; for equivalent functionality in expressions, use the [`concat`](#expressions-concat) and [`get`](#expressions-get) operators.
+   * Value to use for a text label.
```
Comment (anandthakker) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/layers/PropertyFactory.java]: Reflects these changes: https://github.com/mapbox/mapbox-gl-js/pull/6216

**Pull request merged**